app/services/catalog/update_order_item.rb

```diff
@@ -11,22 +11,15 @@ def process
       Rails.logger.info("Processing service order topic message: #{@message} with payload: #{@payload}")
 
       Rails.logger.info("Searching for OrderItem with a task_ref: #{@payload["task_id"]}")
-      order_item = find_order_item
-      Rails.logger.info("Found OrderItem: #{order_item.id}")
+      @order_item = find_order_item
+      Rails.logger.info("Found OrderItem: #{@order_item.id}")
 
-      ManageIQ::API::Common::Request.with_request(order_item.context.transform_keys(&:to_sym)) do
-        order_item.update_message("info", "Task update message received with payload: #{@payload}")
+      ManageIQ::API::Common::Request.with_request(@order_item.context.transform_keys(&:to_sym)) do
+        @order_item.update_message("info", "Task update message received with payload: #{@payload}")
 
-        if @payload["state"] == "completed"
-          order_item.state = "Order Completed"
-          order_item.update_message("info", "Order Complete")
+        @payload["status"] == "ok" ? mark_item_finished : mark_item_failed
```

Comment:
@eclarizio How are dealing with progress messages that are going to show up? That's neither success or failure.